.github/workflows/publish-development.yaml

```diff
@@ -39,22 +39,12 @@ jobs:
         token: ${{ secrets.HUB_JWT }}
         name: zos-${{ github.sha }}.flist
 
-    - name: Promote flist (development)
```

Comment:
You should keep this block. The symlink needs to happen in the `tf-zos` folder not `tf-autobuilder`

Reply:
That become tricky. You don't want tf-zos folder contains 5000 builds...

Reply:
good point. But we need to do something, cause at the moment our code doesn't check anything else the tf-zos folder.
https://github.com/threefoldtech/zos/blob/ee944be1d79e0b021aa24f90bbc5748befa2aac2/bootstrap/etc/zinit/init/bootstrap.sh#L43-L57

Reply:
We can promote directly to `zos:development:latest` which won't be a symlink then, but we don't keep track of the commit id then. We could still do it on tf-autobuilder (so we know the symlink) but not within tf-zos.

Reply:
Maybe development should check into `tf-autobuilder` instead then.

Reply:
> We can promote directly to `zos:development:latest` which won't be a symlink then, but we don't keep track of the commit id then. We could still do it on tf-autobuilder (so we know the symlink) but not within tf-zos.

The problem is we need the symlink to detect which version we are running. The upgraded code does check what the flist points to and parse the name to extract the version.

Reply:
Checking inside `ŧf-autobuilder` seems better IMO then.

Reply:
The name flist `zos:*:latest.flist` must be a symlink, otherwise there is no way to detect the version from  the name and auto-update will never work (basically no way to detect a change in the version). The machine will still boot normally with that flist but basically a new build will never trigger an update.

> Actually even upgraded might misbehave because it won't be able to extract a version number from the name nor the target.

We can change upgraded to just not attempt any updates if no version was detected, but it means development nodes need to restart in case new builds are available. Actually this might be a good idea so your "development/debugging" does not get interrupted when someone else pushes a new change.

Reply:
> We can change upgraded to just not attempt any updates if no version was detected, but it means development nodes need to restart in case new builds are available. Actually this might be a good idea so your "development/debugging" does not get interrupted when someone else pushes a new change.

I do not want to go there. We already have a case where there is not upgrade when a node is not booted from an flist, AKA when this is a local VM from a developer. But we also have some nodes that are long running on real hardware for our devnet. And for these I want to keep the auto update working. This is the first stage of the pipeline that will ensure the upgrade works properly.